err.go

```diff
@@ -629,35 +629,25 @@ func Unwrap(err error) error {
 // constructors
 // ------------------------------------------------------------
 
-// New creates an *Err with the provided Msg.
-//
-// If you have a `ctx` containing other clues data, it is recommended
-// that you call `NewWC(ctx, msg)` to ensure that data gets added to
-// the error.
+// New creates an *Err with the provided Msg.  Ctx can be nil.
 //
 // The returned *Err is an error-compliant builder that can aggregate
 // additional data using funcs like With(...) or Label(...).
-func New(msg string) *Err {
-	return newErr(nil, msg, nil, 1)
+func New(ctx context.Context, msg string) *Err {
```

Comment:
IMO, propagating nil context isn't very good. In a simple func where we don't need ctx, this change could force devs to propagate ctx, where it's actually not needed.

IMO, we should keep both options as separate funcs. This is the usual practice in Go libs and a bunch of other public libs.

IMO, we should simplify clues with other approaches, like moving clues errors to separate pkg `errors`.